src/dataflow/src/source/mod.rs

```diff
@@ -573,15 +546,20 @@ impl ConsistencyInfo {
             // Per partition, we iterate over the data structure to remove (ts,offset) mappings for which
             // we have seen all records <= offset. We keep track of the last "closed" timestamp in that partition
             // in next_partition_ts
-            if let Some(entries) = timestamp_histories.borrow_mut().get_mut(id) {
+            if let Some(entries) = timestamp_histories.borrow().get(&id.source_id) {
```

Comment:
lets see if we can't clean this up / at least mention that the linear scan for BYO sources is subpoptimal and will be fixed soon